src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -214,18 +232,45 @@ impl Coordinator {
 
         match (&config.variant, new_config.variant) {
             (Managed(config), Managed(new_config)) => {
-                self.sequence_alter_cluster_managed_to_managed(
-                    Some(session),
-                    cluster_id,
-                    config,
-                    new_config,
-                    ReplicaCreateDropReason::Manual,
-                )
-                .await?;
+                let needs_finalization = self
+                    .sequence_alter_cluster_managed_to_managed(
+                        Some(session),
+                        cluster_id,
+                        config,
+                        new_config.clone(),
+                        ReplicaCreateDropReason::Manual,
+                        strategy.clone(),
+                    )
+                    .await?;
+                if needs_finalization {
+                    return match strategy.condition {
+                        AlterClusterStrategyCondition::None => Ok(StageResult::Response(
```

Comment:
This return is done in 3 places. I recommend rearranging the code so that the only `return` keyword is only used in the one branch that does something different.